### ISSUE TITLE
`set_payload` in a nested payload

### ISSUE THREAD

**Issue by lambda-science** [qdrant/qdrant-client#780]
Hello.

I tried to change my points payload using Qdrant client. And I face a very non consistent behaviour driving me a bit crazy
My points payload are like:
```json
{
   "payload":{
      "meta":{
         "group_id": "a@b.com",
         "title": "My Document"
      }
   }
}
```

So basically all my fields are in a dict named `meta` and I want to modify the `group_id` with a new value.
I used the following code:
With the following code:
```python
for key, value in uuid_email_mapping.items():
    body = {"group_id": value}
    resp = client.set_payload(
        collection_name=COLLECTION_NAME,
        payload=body,
        key="meta",
        points=qdrant_client.models.Filter(
            must=[
                qdrant_client.models.FieldCondition(
                    key="meta.group_id",
                    match=qdrant_client.models.MatchValue(value=key),
                ),
            ],
        ),
        wait=True,
    )
```

It doesn't work, it put the new group id at the root like
```json
{
   "payload":{
      "group_id": "new group id",
      "meta":{
         "group_id": "a@b.com",
         "title": "My Document"
      }
   }
}
```

With the following code:
```python
for key, value in uuid_email_mapping.items():
    body = {"meta.group_id": value}
    resp = client.set_payload(
        collection_name=COLLECTION_NAME,
        payload=body,
        key="meta",
        points=qdrant_client.models.Filter(
            must=[
                qdrant_client.models.FieldCondition(
                    key="meta.group_id",
                    match=qdrant_client.models.MatchValue(value=key),
                ),
            ],
        ),
        wait=True,
    )
```
It doesn't work, it put the new group id in the meta dict but also with meta prefix, like:
```json
{
   "payload":{
      "meta":{
         "group_id": "a@b.com",
         "title": "My Document",
         "meta.group_id": "new group id",
      }
   }
}
```

**Comment by lambda-science** [comment]
Version of cluster on Qdrant Cloud is 1.11.3 
Related PR: https://github.com/qdrant/qdrant-client/pull/536#issuecomment-1994557600

**Comment by lambda-science** [comment]
```
    body = {'meta.group_id': value}
    resp = client.set_payload(
        collection_name=COLLECTION_NAME,
        payload=body,
        points=qdrant_client.models.Filter(
            must=[
                qdrant_client.models.FieldCondition(
                    key='meta.group_id',
                    match=qdrant_client.models.MatchValue(value=key),
                ),
            ],
        ),
        wait=True,
    )
    print(resp)
```
```json
{
   "payload":{
      "meta.group_id": "new group id",
      "meta":{
         "group_id": "a@b.com",
         "title": "My Document",

      }
   }
}
```
Driving me a bit crazy

**Comment by lambda-science** [comment]
Surprisingly it seem's like it works as expected on localhost but not on qdrant cloud. Very strange ?

**Comment by agourlay** [comment]
Which qdrant-client version are you using? :)

**Comment by lambda-science** [comment]
> Which qdrant-client version are you using? :)
> 
> FYI I cannot reproduce the issue with the REST API on 0.11.x

v1.11.2

Ok maybe it's specific to the client then ? I didn't try with the pure rest API

**Comment by agourlay** [comment]
In this code

```python
for key, value in uuid_email_mapping.items():
    body = {"meta.group_id": value}
    resp = client.set_payload(
        collection_name=COLLECTION_NAME,
        payload=body,
        key="meta",
        points=qdrant_client.models.Filter(
            must=[
                qdrant_client.models.FieldCondition(
                    key="meta.group_id",
                    match=qdrant_client.models.MatchValue(value=key),
                ),
            ],
        ),
        wait=True,
    )
```

the `body` should be `{"group_id": value}` and not `{"meta.group_id": value}`
This is why the prefix `meta` is repeated :)

**Comment by lambda-science** [comment]
MY POINT BEFORE UPADTING (old group id (UUID) is in meta dict.)
![image](https://github.com/user-attachments/assets/73f1b094-5169-43d9-ba75-b60e4c88e9c9)
MY UPDATING CODE (see the key="meta" in the code !)
![image](https://github.com/user-attachments/assets/9dfc0ac0-405d-4d3c-82a2-35adff858cbe)
MY POINT AFTER UPDATING (see the root level group_id ? that's the new value not replacing the one in "META")
![image](https://github.com/user-attachments/assets/cc77a6e1-4df8-4c5a-9452-4598d9e97175)


I swear you will not believe me so I had to make a video out of it because it really doesn't make any sens.
Locally and in your examples everything work fine on my side.
But on my PAID QDRANT CLOUD it really behave differently.
See the screenshots above. Video avaliable for employee only if necessary because it leaks secrets (api key and URL) @agourlay @hh-space-invader

**Comment by lambda-science** [comment]
I HAVE A CLUE:

This behaviour seem's to only happens when a payload index exist on the nested key !!!
![image](https://github.com/user-attachments/assets/8bfa35e1-45bf-46fe-a580-d59cad5a5821)

**Comment by lambda-science** [comment]
Group_id and language both has a payload index (type keyword) on them. When I try to modify them with `set_paypload` with `key="meta"` it doesn't modify them in the `meta` key but put a new payload field at the root of the payload.

For any other nested payload field it works fine (like `embedding_model`, or `embedding_model_id`)
I knew I wasn't crazy !

**Comment by agourlay** [comment]
Sorry, misread the screenshot, this email is your test value.

Can you share the full Python script for reproducing the issue?

**Comment by agourlay** [comment]
We are continuing the investigation on Discord for a faster feedback loop :curly_loop:

**Comment by agourlay** [comment]
fixed in [1.11.5](https://github.com/qdrant/qdrant/releases/tag/v1.11.5)
